temporian/implementation/numpy/data/event.py

```diff
@@ -219,48 +218,24 @@ def to_dataframe(self) -> pd.DataFrame:
 
         Returns:
             pd.DataFrame: DataFrame created from NumpyEvent.
-
         """
+        # Creating an empty dictionary to store the data
+        data = {}
 
-        feature_names = self.feature_names
-        index_names = self.sampling.index
-        columns = index_names + feature_names + ["timestamp"]
-
-        df = pd.DataFrame(data=[], columns=columns)
+        columns = self.sampling.index + self.feature_names + ["timestamp"]
+        for column_name in columns:
+            data[column_name] = []
 
-        # append every feature to the dataframe. without index
         for index, features in self.data.items():
-            timestamps = self.sampling.data[index]
-
-            for i, timestamp in enumerate(timestamps):
-                # add row to dataframe
-                row = (
-                    list(index)
-                    + [feature.data[i] for feature in features]
-                    + [timestamp]
-                )
-                df.loc[len(df)] = row
-
-        # Convert to original dtypes, can be more efficient
-        first_index = self._first_index_value
-        first_features = self._first_index_features
-
-        # get feature dtypes
-        features_dtypes = {
-            feature.name: feature.data[0].dtype for feature in first_features
-        }
-
-        # get tuple index dtypes
-        index_dtypes = {
-            index_name: type(first_index[i])
-            for i, index_name in enumerate(self.sampling.index)
-        }
-
-        # get timestamp dtype
-        first_timestamp = self.sampling.data[first_index][0]
-        sampling_dtype = {"timestamp": first_timestamp.dtype}
-
-        df = df.astype({**features_dtypes, **index_dtypes, **sampling_dtype})
+            timesamples = self.sampling.data[index]
+            data["timestamp"].extend(timesamples)
+            for feature in features:
+                data[feature.name].extend(feature.data)
+            for index_key in self.sampling.index:
+                data[index_key].extend(index * len(timesamples))
+
+        # Converting dictionary to pandas DataFrame
+        df = pd.DataFrame(data)
```

Comment:
Does this keep the original `NumpyEvent's` index? i.e. is the output `DataFrame` indexed by the same columns as the event?

Reply:
We talked about it and decided that it was best for the output DataFrame to not have any index